### PR TITLE
Remove obsolete `OrEqualTo` methods

### PR DESCRIPTION
**Pull request by IT-VBFK**
This refs: #1677 

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [x] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [x] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Review by dennisdoomen — APPROVED** [review]

**Review by jnyrup — COMMENTED** [review]
Thanks 👍 

Here's a few more places that needs to be changed as well.

```
docs/_data/mstest-migration/assert.yml:249:    actual.Should().BeLessOrEqualTo(expected);
docs/_data/mstest-migration/assert.yml:261:    actual.Should().BeGreaterOrEqualTo(expected);
docs/_data/mstest-migration/assert.yml:297:    actual.Should().BeGreaterOrEqualTo(expected);
docs/_data/mstest-migration/assert.yml:309:    actual.Should().BeLessOrEqualTo(expected);
docs/_data/tips/collections.yml:123:    actual.Count().Should().BeGreaterOrEqualTo(k);
docs/_data/tips/collections.yml:126:    actual.Should().HaveCountGreaterOrEqualTo(k);
docs/_data/tips/collections.yml:147:    actual.Count().Should().BeLessOrEqualTo(k);
docs/_data/tips/collections.yml:150:    actual.Should().HaveCountLessOrEqualTo(k);
```

```
Src/FluentAssertions/Numeric/NumericAssertions.cs:230:    public AndConstraint<TAssertions> BeLessOrEqualTo
Src/FluentAssertions/Numeric/NumericAssertions.cs:292:    public AndConstraint<TAssertions> BeGreaterOrEqualTo
```

**Comment by dennisdoomen** [comment]
> Here's a few more places that needs to be changed as well.

You can always count on @jnyrup to be thorough 😜💪

File: docs/_pages/releases.md [hunk 1]
```diff
@@ -16,6 +16,19 @@ sidebar:
...
 
 ### Breaking Changes (for users)
+* Removed obsolete `...OrEqualTo` methods - [#2269](https://github.com/fluentassertions/fluentassertions/pull/2269)
+  * `GenericCollectionAssertions`
+    * `HaveCountGreaterOrEqualTo`: Use `HaveCountGreaterThanOrEqualTo` 
+    * `HaveCountLessOrEqualTo`: Use `HaveCountLessThanOrEqualTo` 
+  * `ComparableTypeAssertions`
+    * `BeGreaterOrEqualTo`: Use `BeGreaterThanOrEqualTo`
+    * `BeLessOrEqualTo`: Use `BeLessThanOrEqualTo`
+  * `SimpleTimespanAssertions`
```
Comment (jnyrup) [inline docs/_pages/releases.md:26]:
```suggestion
  * `SimpleTimeSpanAssertions`
```

**Comment by IT-VBFK** [comment]
Can anybody get this Qodana issues? 😂 

This files don't exist anymore...

**Review by jnyrup — APPROVED** [review]

**Pull request merged**